src/Domain/Infra/Doctrine/MappingConfig.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MsgPhp\Domain\Infra\Doctrine;
+
+/**
+ * @author Pascal Wacker <hello@pascalwacker.ch>
+ *
+ * @internal
+ */
+final class MappingConfig
+{
+    public $mappingFiles;
+    public $mappingDir;
+    public $keyMaxLength;
+
+    public function __construct(array $mappingConfig = [])
```

Comment:
ok, let's change plans a bit: `___construct(array $mappingFiles, array $mappingConfig)`

to be wired with `%msgphp.doctrine.mapping_files%` and `%msgphp.doctrine.mapping_config%` respectively

Reply:
then we can just always set $mappingFiles and do `if (isset($mappingConfig['key_max_length'])` per option, and move the defaults to the properties.